sky/serve/serve_utils.py

```diff
@@ -709,6 +712,129 @@ def _get_replica_status() -> serve_state.ReplicaStatus:
     return ''
 
 
+def has_valid_replica_id(file_name: str,
+                         target_replica_id: Optional[int]) -> bool:
+    if target_replica_id is None:
+        return True
+
+    match = re.search(serve_constants.REPLICA_ID_PATTERN, file_name)
+
+    if match is not None:
+        replica_id = int(match.group(1))
+    else:
+        raise ValueError(
+            f'Failed to get replica id from file name: {file_name}')
+
+    return replica_id == target_replica_id
+
+
+def prepare_replica_logs_for_download(service_name: str, timestamp: str,
```

Comment:
I'm a little bit confused about the complicated logic in this function. Can we just construct all possible file name and check if them exist? Why do we need to looping through all replica log files?

https://github.com/skypilot-org/skypilot/blob/c6ae536d8dfedc3bbcf427a81480382b9d5f4c29/sky/serve/serve_utils.py#L227-L237

Reply:
Yes, actually I didn't notice that the two functions were already encapsulated, so I initially took the traversal approach, which was quite cumbersome. Improvements have been made now. Please kindly review :)